scipp-imaging-2D.ipynb

```diff
@@ -0,0 +1,8350 @@
+    "    \"\"\"\n",
+    "    tof_values = list()\n",
+    "    with open(tof_file) as fh:\n",
+    "        csv_reader = csv.reader(fh, delimiter='\\t')\n",
+    "        next(csv_reader, None)  # skip header\n",
+    "        for row in csv_reader:\n",
+    "            tof_values.append(float(row[1]))\n",
+    "    return tof_values\n",
+    "\n",
+    "\n",
+    "def _load_tiffs(tiff_dir):\n",
+    "    if not os.path.isdir(tiff_dir):\n",
+    "        raise RuntimeError(tiff_dir + \" is not directory\")\n",
+    "    stack = []\n",
+    "    path_length = len(tiff_dir) + 1\n",
+    "    filenames = sorted(glob.glob(tiff_dir + \"/*.tiff\"))\n",
+    "    nfiles = len(filenames)\n",
+    "    count = 0\n",
+    "    print(f\"Loading {nfiles} files from '{tiff_dir}'\")\n",
+    "    for filename in filenames:\n",
+    "        count += 1\n",
+    "        print('\\r{0}: Image {1}, of {2}'.format(filename[path_length:], count, nfiles), end=\"\")\n",
+    "        img = fabio.open(os.path.join(tiff_dir, filename))\n",
+    "        stack.append(np.flipud(img.data))\n",
+    "\n",
+    "    return np.array(stack)\n",
+    "\n",
+    "def tiffs_to_variable(tiff_dir):\n",
+    "    \"\"\"\n",
+    "    Loads all tiff images from the directory into a scipp Variable.\n",
+    "    \"\"\"\n",
+    "    stack = _load_tiffs(tiff_dir)\n",
+    "    return sc.Variable([Dim.Tof, Dim.Spectrum], \n",
+    "                       values=stack.astype(np.float64).reshape(\n",
+    "                           stack.shape[0], stack.shape[1]*stack.shape[2]))\n",
+    "\n",
+    "from functools import reduce\n",
+    "\n",
+    "def stitch(data_array, frame_parameters, rebin_parameters):\n",
+    "    \"\"\"\n",
+    "    Stitches the 5 different frames data.\n",
+    "    \n",
+    "    It crops out each frame, then shifts it so that all frames align,\n",
+    "    and then rebins to the common bins used for all frames.\n",
+    "    \"\"\"\n",
+    "    frames = []\n",
+    "\n",
+    "    rebin_params = sc.Variable([Dim.Tof], values=np.arange(*rebin_parameters, dtype=np.float64))\n",
+    "    \n",
+    "    for i, (slice_bins, shift_parameter) in enumerate(zip(frame_parameters, frame_shifts)):\n",
+    "        bins = sc.Variable([Dim.Tof], values=np.arange(*slice_bins, dtype=np.float64))\n",
+    "        # Rebins the whole data to crop it to frame bins\n",
+    "        rebinned = sc.rebin(data_array, Dim.Tof, bins)\n",
+    "        # Shift the frame backwards to make all frames overlap\n",
+    "        rebinned.coords[Dim.Tof] += shift_parameter\n",
+    "        # Rebin to overarching coordinates so that the frame coordinates align\n",
+    "        rebinned = sc.rebin(rebinned, Dim.Tof, rebin_params)\n",
+    "\n",
+    "        frames.append(rebinned)\n",
+    "\n",
+    "    for f in frames[1:]:\n",
+    "        frames[0] += f\n",
+    "\n",
+    "    return frames[0]\n",
+    "\n",
+    "def remove_special_values(values):\n",
```

Comment:
Shouldn't this be `zero_special_values`?

Reply:
I have renamed it as that name does make more sense.